pkg/schedule/schedulers/hot_region.go

```diff
@@ -789,12 +794,17 @@ func (bs *balanceSolver) filterSrcStores() map[uint64]*statistics.StoreLoadDetai
 			continue
 		}
 
-		if bs.checkSrcByPriorityAndTolerance(detail.LoadPred.Min(), &detail.LoadPred.Expect, srcToleranceRatio) {
-			ret[id] = detail
-			hotSchedulerResultCounter.WithLabelValues("src-store-succ", strconv.FormatUint(id, 10)).Inc()
-		} else {
-			hotSchedulerResultCounter.WithLabelValues("src-store-failed", strconv.FormatUint(id, 10)).Inc()
+		if !bs.checkSrcByPriorityAndTolerance(detail.LoadPred.Min(), &detail.LoadPred.Expect, srcToleranceRatio) {
+			hotSchedulerResultCounter.WithLabelValues("src-store-failed-"+bs.resourceTy.String(), strconv.FormatUint(id, 10)).Inc()
+			continue
 		}
+
+		if !bs.checkSrcHistoryLoadsByPriorityAndTolerance(&detail.LoadPred.Current, &detail.LoadPred.Expect, srcToleranceRatio) {
```

Comment:
This will slow down the convergence speed of hotspot scheduling. Have you compared the previous cases with small region? it's expect stable at lease 5mins now. 
BTW, for conservative, we can only do this check in large regions.

Reply:
good guggestion, return loads direct.

Reply:
Q1: Have you compared the previous cases with small region?
yes, the issue has shown that `hot-region-schduler` brings some useless scheduler in small region. 
Q2: it's expect stable at lease 5mins now.
not yet, PD can scheduler the source or target store if the store load is higher or lower than the average continue. the scheduled store's loads can be unstable in this 5min.
Q3:  we can only do this check in large regions?
maybe, I should one configuration to switch this check.

Reply:
I have add condition only for the `raftkv2`, so the pr has no influence for the `raftkv1`.